coderd/workspaces_test.go

```diff
@@ -426,47 +426,441 @@ func TestWorkspace(t *testing.T) {
 
 	t.Run("TemplateVersionPreset", func(t *testing.T) {
 		t.Parallel()
-		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
-		user := coderdtest.CreateFirstUser(t, client)
-		authz := coderdtest.AssertRBAC(t, api, client)
-		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
-			Parse: echo.ParseComplete,
-			ProvisionPlan: []*proto.Response{{
-				Type: &proto.Response_Plan{
-					Plan: &proto.PlanComplete{
-						Presets: []*proto.Preset{{
-							Name: "test",
-						}},
+
+		testCases := []struct {
+			name                      string
+			presets                   []*proto.Preset
+			templateVersionParameters []*proto.RichParameter
+			selectedPresetIndex       int // -1 if no preset should be used
+		}{
+			{
+				name:                      "No Presets",
+				presets:                   []*proto.Preset{},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       -1,
+			},
+			{
+				name: "Single Preset - No Parameters",
+				presets: []*proto.Preset{{
+					Name: "test",
+				}},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       0,
+			},
+			{
+				name: "Single Preset - With Parameters But No Template Parameters",
```

Comment:
You don't have a case here for the inverse, where you have template parameters but no preset params chosen, which is the most common case I'd argue.

Reply:
done